pkg/vsphere/vm/vm.go

```diff
@@ -281,7 +282,27 @@ func (vm *VirtualMachine) DeleteExceptDisks(ctx context.Context) (*object.Task,
 		return nil, err
 	}
 
-	return vm.Destroy(op)
+	info, err := vm.WaitForResult(op, func(ctx context.Context) (tasks.Task, error) {
+		return vm.Destroy(ctx)
+	})
+
+	if err == nil || !tasks.IsMethodDisabledError(err) {
+		return info, err
+	}
+
+	// If destroy method is disabled on this VM, re-enable it and retry
+	if tasks.IsMethodDisabledError(err) {
+		err = vm.EnableDestroy(ctx)
```

Comment:
Should this follow the `tasks.IsConcurrentAccessError` retry pattern?

Reply:
Different callers of `DeleteExceptDisks` treat `ConcurrentAccess` failure differently - when portlayer calls it to remove cVM, it ignores the `ConcurrentAccess` failure because it shouldn't occur given that we're not setting ChangeVersion. 
https://github.com/vmware/vic/blob/master/lib/portlayer/exec/container.go#L639
The retry on `ConcurrentAccess` error is probably not necessary but since we're removing appliance VM and all of the cVMs, we probably want to make sure everything is gone

Reply:
> not enough permission to enable the operation due to some ops-user creds configuration stuff, corrupted VM, etc.

It seems like these examples are all be different errors (not Concurrent Access Error).

If there's just a concurrent access error, why wouldn't we want to just retry?

Reply:
If `ConcurrentAccess` error is returned from `vm.Destroy`, it makes more sense for the caller to decide what to do (like portlayer ignores it because it's not setting `ChangeVersion`, and vic-machine retries on it)
If `ConcurrentAccess` is returned from `vm.EnableDestroy`, then it probably needs a retry, since we should at least grant us the permission to destroy it first
Fixing it, thanks